src/cryptoadvance/specter/server_endpoints/settings.py

```diff
@@ -279,8 +283,9 @@ def tor():
                 tor_connectable = res.status_code == 200
                 if tor_connectable:
                     flash("Tor requests test completed successfully!", "info")
-                    logger.error("Tor-Logs:")
-                    logger.error(app.specter.tor_daemon.get_logs())
+                    if tor_type == "builtin":
+                        logger.error("Tor-Logs:")
+                        logger.error(app.specter.tor_daemon.get_logs())
```

Comment:
This won't work if tor is running properly. If it's not running properly, it's no wonder the test failed.
So maybe the best move here is stopping tor, getting the logs and start it again?!